### ISSUE TITLE
`no-unnecessary-type-arguments` false positive in v0.20.0

### ISSUE THREAD

**Issue by Perdolique** [oxc-project/tsgolint#875]
<details><summary>test.ts</summary>
<p>

```ts
import {
  useSession,
  createError,
  type H3Event,
  type EventHandlerRequest,
  type SessionConfig
} from 'h3'


interface SessionData {
  userId?: string;
}

function getSessionConfig() : SessionConfig {
  return {
    password: '',
    name: '',

    cookie: {
      sameSite: 'strict',
      httpOnly: true,
      secure: true
    }
  }
}

async function useAppSession(event: H3Event<EventHandlerRequest>) {
  const config = getSessionConfig()

  return useSession<SessionData>(event, config)
}

async function validateSessionUser(event: H3Event<EventHandlerRequest>) {
  const session = await useAppSession(event)
  const { userId } = session.data

  if (userId === undefined) {
    throw createError({
      status: 401
    })
  }

  return userId
}

export { useAppSession, validateSessionUser }
```

</p>
</details> 

Code above triggers an error with `oxlint-tsgolint` `v0.20.0` (`v0.19.0` works correctly):

```text
  × typescript-eslint(no-unnecessary-type-arguments): This is the default value for this type parameter, so it can be omitted.
    ╭─[server/utils/test.ts:30:21]
 29 │ 
 30 │   return useSession<SessionData>(event, config)
    ·                     ───────────
 31 │ }
    ╰────
```

`SessionData` can't be omitted, because it provides typed `data`. W/o it, the `data` is `any`.

**Comment by camc314** [comment]
we need to copy upstream and revert this change: https://github.com/typescript-eslint/typescript-eslint/commit/e372a6697de3fea6e5b635c72177df7cf697f2ce

**Comment by camc314** [comment]
Hi, apologies for the delay.

v0.21.1 is releasing now.

Please let me know if it does not address this issue

Thanks!

**Comment by Perdolique** [comment]
@camc314, it's still the problem in oxlint v1.60.0 and oxlint-tsgolint v0.21.1. Should oxlint be updated as well?

**Comment by camc314** [comment]
should just have to upgrade oxlint-tsgolint.

i was pretty sure this was fixed, will take another look

**Comment by Perdolique** [comment]
@camc314 [reproduction ](https://github.com/Perdolique/tsgolint-bug-875), just in case.

**Comment by camc314** [comment]
Apologies for the wait @Perdolique this rule has been a pain to get right!
